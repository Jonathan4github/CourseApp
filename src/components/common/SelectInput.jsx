import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  defaultOption: PropTypes.string,
  value: PropTypes.string,
  error: PropTypes.string,
  options: PropTypes.arrayOf(PropTypes.shape())
};

const defaultProps = {
  value: '',
  error: '',
  defaultOption: '',
  options: []
};

const SelectInput = ({
  name, label, onChange, defaultOption, value, error, options
}) => (
  <div className="form-group">
    <div className="field">
      <label htmlFor={name}>{label}</label>
      <select
      className="custom-select"
      name={name}
      value={value}
      onChange={onChange}>
        <option value="">{defaultOption}</option>
        {options.map(option => (
          <option key={option.id} value={option.id}>
            {`${option.firstName} ${option.lastName}`}
          </option>
        ))}
      </select>
      {error && <div className="alert alert-danger">{error}</div>}
    </div>
  </div>
);

SelectInput.propTypes = propTypes;
SelectInput.defaultProps = defaultProps;

export default SelectInput;
