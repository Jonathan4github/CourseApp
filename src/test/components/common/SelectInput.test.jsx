/* eslint-disable */
import SelectInput from '../../../components/common/SelectInput';

const setup = (error) => {
  const props = {
    name: '',
    label: '',
    onChange: jest.fn(),
    onFocus: jest.fn(),
    defaultOption: '',
    value: '',
    error,
    options: [{ id: '', firstName: '', lastName: '' }]
  };

  return mount(<SelectInput {...props} />);
};

let mountWrapper;

describe('<SelectInput /> component test', () => {
  it('renders component without crashing', () => {
    mountWrapper = setup('');
    expect(toJson(mountWrapper)).toMatchSnapshot();
  });

  it('should not render error divs when no error occur', () => {
    mountWrapper = setup('');
    expect(mountWrapper.find('.has-error').length).toEqual(0);
    expect(mountWrapper.find('.field-error').length).toEqual(0);
  });

  it('renders error divs when error occur', () => {
    mountWrapper = setup('Select an author');

    expect(mountWrapper.find('.has-error')).toBeDefined();
    expect(mountWrapper.find('.has-error').length).toEqual(1);
    expect(mountWrapper.find('.field-error')).toHaveLength(1);
    expect(mountWrapper.find('.field-error').get(0).props.children).toEqual('Select an author');
  });
});
