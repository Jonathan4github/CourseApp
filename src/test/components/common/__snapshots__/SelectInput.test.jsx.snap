// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectInput /> component test renders component without crashing 1`] = `
<SelectInput
  defaultOption=""
  error=""
  label=""
  name=""
  onChange={[MockFunction]}
  onFocus={[MockFunction]}
  options={
    Array [
      Object {
        "firstName": "",
        "id": "",
        "lastName": "",
      },
    ]
  }
  value=""
>
  <div
    className="form-group"
  >
    <div
      className="field"
    >
      <label
        htmlFor=""
      />
      <select
        className="custom-select"
        name=""
        onChange={[MockFunction]}
        onFocus={[MockFunction]}
        value=""
      >
        <option
          value=""
        />
        <option
          key=""
          value=""
        >
           
        </option>
      </select>
    </div>
  </div>
</SelectInput>
`;
